src/IRI.php

```diff
@@ -301,12 +301,9 @@ protected function parse_iri(string $iri)
             if ($match[1] === '') {
                 $match['scheme'] = null;
             }
-            if (!isset($match[3]) || $match[3] === '') {
+            if ($match[3] === '') {
                 $match['authority'] = null;
             }
-            if (!isset($match[5])) {
```

Comment:
Looks good, `$match[3]` will never be unset since `$match[5]` is matched unconditionally:
https://3v4l.org/vifvK